src/app/services/shopping-cart.service.ts

```diff
@@ -41,4 +41,10 @@ export class ShoppingCartService {
   isOnCart(code: string): boolean {
     return this.shoppingCart.filter(product => product.code === code).length > 0;
   }
+
+  clearShoppingCart(): void {
+    this.localStorageService.clearLocalStorage();
+    this.shoppingCart = [];
+  }
```

Comment:
Trigger the subject to reload the state on the app, and avoid the location reload that you use before.
```suggestion
  clearShoppingCart(): void {
    this.localStorageService.clearLocalStorage();
    this.shoppingCart = [];
    this.subject$.next('');
  }
```